test/e2e/apimachinery/crd_conversion_webhook.go

```diff
@@ -461,17 +461,17 @@ func testCRListConversion(ctx context.Context, f *framework.Framework, testCrd *
 	list, err := customResourceClients["v1"].List(ctx, metav1.ListOptions{})
 	framework.ExpectNoError(err)
 	gomega.Expect(len(list.Items)).To(gomega.BeIdenticalTo(2))
-	framework.ExpectEqual((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
-		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1), true)
+	gomega.Expect((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
+		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1)).To(gomega.BeTrue())
```

Comment:
This is basically checking that a list contains two elements in arbitrary order, and identifies the elements by their name. What we want to see if this fails is the `list.Items`.

Does this work?
```
Expect(list.Items).To(ConsistOf(HaveField("GetName()", Equal(name1)), HaveField("GetName()", Equal(name2)))
```

Checking the length separately is not necessary when using `ConsistOf`.

Reply:
@pohly It does not work as expected. please see

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119459/pull-kubernetes-e2e-kind-ipv6/1698967903245701120

Reply:
Well, it was worth a try :sweat_smile:  Please revert back to the more complicated separate calls.

Looks like an issue with Gomega (cannot call GetName?). I'll try to reproduce and file an issue.

Reply:
@pohly I pushed some changes, please review again. 

https://github.com/kubernetes/kubernetes/compare/01905fe68da8efcc714122aed38507a2787a414d..15a402eefb7258f05da6a8bcb807ebe129f326d1